src/compiler/Restler.Compiler/SwaggerVisitors.fs

```diff
@@ -174,96 +177,105 @@ module SwaggerVisitors =
-                                Constant (primitiveType, formatJTokenProperty primitiveType v)
-                            | _ -> raise (invalidOp(sprintf "invalid payload %A, expected fuzzable" propertyValue))
-
-                        LeafNode { propertyPayload with
-                                      payload = examplePayload }
-
-                | NJsonSchema.JsonObjectType.None ->
-                    // When the object type is 'None', simply pass through the example value ('propertyValue')
-                    // For example: the schema of a property whose schema is declared as a $ref will be visited here.
-                    // The example property value needs to be examined according to the 'ActualSchema', which
-                    // is passed through below.
 
-                    let tree = generateGrammarElementForSchema property.ActualSchema propertyValue parents
+    let rec processProperty (propertyName, property:NJsonSchema.JsonSchemaProperty)
+                        (propertyValue: JToken option)
+                        (parents:NJsonSchema.JsonSchema list)
+                        (cont: Tree<LeafProperty, InnerProperty> -> Tree<LeafProperty, InnerProperty>) =
+        // 'isReadOnly' is not correctly initialized in NJsonSchema.  Instead, it appears
+        // in ExtensionData
+        let propertyIsReadOnly (property:NJsonSchema.JsonSchemaProperty) =
+            match getExtensionDataBooleanPropertyValue property.ExtensionData "readOnly" with
+            | None -> property.IsReadOnly
+            | Some v -> v
+
+        match property.Type with
+            | NJsonSchema.JsonObjectType.String
+            | NJsonSchema.JsonObjectType.Number
+            | NJsonSchema.JsonObjectType.Integer
+            | NJsonSchema.JsonObjectType.Boolean ->
+                let propertyPayload =
+                    let propertySchema = (property :> NJsonSchema.JsonSchema)
+
+                    getFuzzableValueForProperty propertyName
+                             propertySchema
+                             property.IsRequired
+                             (propertyIsReadOnly property)
+                             (tryGetEnumeration propertySchema)
+                             (tryGetDefault propertySchema)
+
+                match propertyValue with
+                | None -> cont(LeafNode propertyPayload)
+                | Some v ->
+                    let examplePayload =
+                        match propertyPayload.payload with
+                        | Fuzzable (primitiveType, _) ->
+                            // Replace the default payload with the example payload, preserving type information.
+                            Constant (primitiveType, GenerateGrammarElements.formatJTokenProperty primitiveType v)
+                        | _ -> raise (invalidOp(sprintf "invalid payload %A, expected fuzzable" propertyValue))
+
+                    cont(LeafNode { propertyPayload with payload = examplePayload })
```

Comment:
>cont [](start = 20, length = 4)

This alternative style seems more readable to me as compared with the original code.
                    LeafNode propertyPayload
                    |> cont // on separate line